src/distributions/distribution-factory.ts

```diff
@@ -25,6 +28,9 @@ export function getJavaDistribution(
       return new AdoptDistribution(installerOptions, AdoptImplementation.Hotspot);
     case JavaDistribution.AdoptOpenJ9:
       return new AdoptDistribution(installerOptions, AdoptImplementation.OpenJ9);
+    case JavaDistribution.Adoptium:
+    case JavaDistribution.AdoptiumHotspot:
+      return new AdoptiumDistribution(installerOptions, AdoptiumImplementation.Hotspot);
```

Comment:
return new AdoptiumDistribution(installerOptions, AdoptiumImplementation.Hotspot);

should be 

return new TemurinDistribution(installerOptions, TemurinImplementation);
(or maybe can be simplified here and in the callee to drop the TemurinImplementation parameter, as there is probably no need (three is no openj9 variant to consider).